### PR TITLE
De-alias symbols similar to tsserver's go-to-definition.

### PR DESCRIPTION
**Pull request by varungandhi-src**
Fixes issue #46.

In the first commit, I've added a generated snapshot to show that we add the
incorrect reference. The fix is implemented in the second commit, where I'm using
logic from tsserver to make sure we de-alias symbols (which appear in the
presence of imports). So the generated snapshots now have the right references.

I've left some TODOs and FIXMEs since there are some edge cases around
JS files and dynamic imports which are handled by tsserver, and handling those
required inlining a bunch of more code. It wasn't super clear to me how all
those edge cases should be tested, so I've avoided inlining those.
I will file a separate follow-up issue to address those TODOs.

### Test plan

Added snapshot tests.

### CODE REVIEW

File: src/TypeScriptInternal.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  switch (node.kind) {
+    case ts.SyntaxKind.ImportClause:
+    case ts.SyntaxKind.ImportEqualsDeclaration:
+      // TODO: How do we test this code path?
+      return true;
+    case ts.SyntaxKind.ImportSpecifier:
+      return node.parent.kind === ts.SyntaxKind.NamedImports;
+    case ts.SyntaxKind.BindingElement:
+    case ts.SyntaxKind.VariableDeclaration:
+      // TODO: How do we test this code path?
```
Comment (Strum355) [inline src/TypeScriptInternal.ts:16]: Able to put just a standard JS file alongside the other sources?

Comment (varungandhi-src) [reply src/TypeScriptInternal.ts:16]: That's not enough, we need to somehow create a BindingElement or VariableDeclaration that is also an alias and I don't know how to do that, because the different `SymbolFlag` values and syntax node types in the TypeScript compiler aren't documented.

**Review by Strum355 — APPROVED** [review]
